src/AI/Muriel.java

```diff
@@ -138,24 +141,10 @@ private UnitMatchupAndMetaInfo getUnitMatchupInfo(Unit myUnit, Unit otherUnit)
     UnitMatchupAndMetaInfo umami = myUnitEffectMap.get(new UnitModelPair(myUnit.model, otherUnit.model));
     if( null != umami ) return umami;
 
-    double myDamage = 0;
-    WeaponModel myWeapon = myUnit.chooseWeapon(otherUnit.model, 1, myUnit.model.hasMobileWeapon());
-    if( null != myWeapon )
-    {
-      BattleInstance.BattleParams params = new BattleInstance.BattleParams(myUnit, myWeapon,
-          otherUnit, Environment.getTile(TerrainType.ROAD, Weathers.CLEAR), false, null);
-      myDamage = params.calculateDamage();
-    }
+    double myDamage = CombatEngine.buildSimpleAttack(myUnit, 1, otherUnit, myCo.myView, 0, myUnit.model.hasMobileWeapon()).calculateDamage();
 
     // Now go the other way.
-    double otherDamage = 0;
-    WeaponModel otherWeapon = otherUnit.chooseWeapon(myUnit.model, 1, myUnit.model.hasMobileWeapon());
-    if( null != otherWeapon )
-    {
-      BattleInstance.BattleParams params = new BattleInstance.BattleParams(otherUnit, otherWeapon,
-          myUnit, Environment.getTile(TerrainType.ROAD, Weathers.CLEAR), false, null);
-      otherDamage = params.calculateDamage();
-    }
+    double otherDamage = CombatEngine.buildSimpleAttack(otherUnit, 1, myUnit, myCo.myView, 0, false).calculateDamage();
```

Comment:
This is certainly cleaner (except or magic numbers aaargh). Nice.

Reply:
Exactly as many magic numbers as before!